lib/typescript.js

```diff
@@ -55,6 +55,18 @@ function convertType(swaggerType, swagger) {
             var property = convertType(propertyType);
             property.name = propertyName;
             typespec.properties.push(property);
+            // see if property is required
```

Comment:
It is. I think you have the correct understanding.

One goal was to give in response type information to the templates.

The other goal was to add the information about optional properties